x-pack/plugins/stack_alerts/server/rule_types/es_query/executor.ts

```diff
@@ -48,26 +49,28 @@ export async function executor(core: CoreSetup, options: ExecutorOptions<EsQuery
   // of the rule, the latestTimestamp will be used to gate the query in order to
   // avoid counting a document multiple times.
 
-  const { numMatches, searchResult, dateStart, dateEnd } = esQueryRule
-    ? await fetchEsQuery(ruleId, name, params as OnlyEsQueryRuleParams, latestTimestamp, {
-        scopedClusterClient,
-        logger,
-      })
+  const { numMatches, searchResult, dateStart, dateEnd, link } = esQueryRule
+    ? await fetchEsQuery(
+        ruleId,
+        name,
+        params as OnlyEsQueryRuleParams,
+        latestTimestamp,
+        base,
+        spacePrefix,
+        {
+          scopedClusterClient,
+          logger,
+        }
+      )
     : await fetchSearchSourceQuery(ruleId, params as OnlySearchSourceRuleParams, latestTimestamp, {
```

Comment:
Should this take spaceId into account?

Reply:
Shouldn't locator service take care of this?

Reply:
I see the link show up as `/app/r?l=DISCOVER_APP_LOCATOR&v=8.7.0&` even for a rule in a different space. Going to the link takes me back to the default space. I see results showing up in Discover though. Is that how it's supposed to work?

Reply:
Fixed